bdk-ffi/src/tx_builder.rs

```diff
@@ -237,6 +248,10 @@ impl TxBuilder {
         if let Some(sequence) = self.sequence {
             tx_builder.set_exact_sequence(Sequence(sequence));
         }
+        if !&self.data.is_empty() {
+            let push_bytes = PushBytesBuf::try_from(self.data.clone())?;
+            tx_builder.add_data(&push_bytes);
+        }
```

Comment:
I'm reviewing these commit by commit as recommended, and for this `add_data` I was wondering if we needed to manually validate the size of the data here based on Bitcoin's standard transaction relay policy. I'm wondering this because that thought popped into my head so I started writing some tests around it like:
- `test_add_data`
- `test_add_data_within_limit`
- `test_add_data_exceeding_limit`
- `test_add_empty_data`

and I couldn't get `fn test_add_data_exceeding_limit` test to work without adding something like 
```
            if self.data.len() > 80 {
                return Err(CreateTxError::PushBytesError);
            }
```

in the `finish` method.

Thoughts? I could totally be missing something though.

Also let me know if adding the specific implementation of my tests is helpful too.

Reply:
Personally, I don't think `add_data` should land for `1.0.0`. `PushBytesBuf` doesn't have a clear API and `is_standard_op_return` isn't even a method yet in `rust-bitcoin` (I am [patching](https://github.com/rust-bitcoin/rust-bitcoin/pull/3643) the method right now). Ideally everything about OP_RETURN is left to the Rust-side and FFI has no logic for handling this

Reply:
> Ideally everything about OP_RETURN is left to the Rust-side and FFI has no logic for handling this

Def agree with this 100%. I just happened to be running into an issue with it when I was coming up with test to test what happened when I gave it data exceeding that limit. But maybe I wrote my test incorrectly, will double back and look at it again

Reply:
There is no check on the Rust side and currently it will allow for invalid OP_RETURN data

Reply:
Looking back at my test the other thing that played into the way I wrote it was how I approached non-standard transactions.

Reply:
> There is no check on the Rust side and currently it will allow for invalid OP_RETURN data

Thanks for this! I didn't see it before I wrote:

> Looking back at my test the other thing that played into the way I wrote it was how I approached non-standard transactions.

I'm actually really interested in where we land on this conversation even if for the time being overall we end up at this place you mentioned rob:

> I don't think add_data should land for 1.0.0